### PR TITLE
LPS-101098 Create an API method to search for users assigned to an account

### PR DESCRIPTION
**Pull request by drewbrokke**
This is an update for [LPS-101098](https://issues.liferay.com/browse/LPS-101098).<br><br>/cc @pei-jung @alee8888 @ChrisKian @dejuknow

### CODE REVIEW

**Comment by drewbrokke** [comment]
ci:test:sf

**Comment by drewbrokke** [comment]
ci:test:relevant

File: modules/apps/account/account-service/src/main/java/com/liferay/account/internal/search/search/spi/model/query/contributor/UserModelPreFilterContributor.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+/**
+ * @author Drew Brokke
+ */
+@Component(
+	immediate = true,
+	property = "indexer.class.name=com.liferay.portal.kernel.model.User",
+	service = ModelPreFilterContributor.class
+)
+public class UserModelPreFilterContributor
```
Comment (pei-jung) [inline modules/apps/account/account-service/src/main/java/com/liferay/account/internal/search/search/spi/model/query/contributor/UserModelPreFilterContributor.java]: should we prefix with Account so `AccountUserModelPreFilterContributor`?

File: modules/apps/account/account-service/src/main/java/com/liferay/account/internal/search/spi/model/index/contributor/UserModelDocumentContributor.java [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+/**
+ * @author Drew Brokke
+ */
+@Component(
+	immediate = true,
+	property = "indexer.class.name=com.liferay.portal.kernel.model.User",
+	service = ModelDocumentContributor.class
+)
+public class UserModelDocumentContributor
```
Comment (pei-jung) [inline modules/apps/account/account-service/src/main/java/com/liferay/account/internal/search/spi/model/index/contributor/UserModelDocumentContributor.java]: prefix with Account?

File: modules/apps/account/account-service/src/main/java/com/liferay/account/internal/retriever/AccountUserRetrieverImpl.java [hunk 3]
```diff
@@ -40,9 +60,136 @@
...
+
+				attributes.put(
+					"accountEntryIds",
+					new long[] {accountEntry.getAccountEntryId()});
+				attributes.put("params", new LinkedHashMap<>());
+				attributes.put("status", status);
+
+				searchContext.setAttributes(attributes);
+
+				searchContext.setCompanyId(accountEntry.getCompanyId());
```
Comment (pei-jung) [inline modules/apps/account/account-service/src/main/java/com/liferay/account/internal/retriever/AccountUserRetrieverImpl.java]: Extracting this to a helper method might help with readability and follow the pattern in UserLocalServiceImpl.buildSearchContext

**Comment by drewbrokke** [comment]
https://github.com/brianchandotcom/liferay-portal/pull/79167

**Pull request closed without merging**